### PR TITLE
Mutexpend update references

### PR DESCRIPTION
**Pull request by ray4477**
# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [x] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [x] Software feature has test associated with it. 
- [x] Software feature has documentation for it updated.
- [x] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [x] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [x] Did you Have Fun?

## Things to Consider 
- [x] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

**Review by connorl309 — APPROVED** [review]
Don't see any issues here with the replacements. Not sure how we'll write test files for RTOS changes though.

**Comment by ClarkPoon** [comment]
> # Quality Assurance Checklist
> 
> To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._
> 
> There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions.
> ## Requirements
> 
>     * [x]  Followed Coding Style Guide
> 
>     * [x]  Code Build checks pass
> 
>     * [x]  No merge conflicts
> 
>     * [x]  Software feature has test associated with it.
> 
>     * [x]  Software feature has documentation for it updated.
> 
>     * [x]  Test provides useful information and uses relevant data to accurately represent BPS
>       
>       * NOTE: If test file already exists, use that one
> 
>     * [x]  2 Members of the team have already reviewed my PR
>       
>       * NOTE: Approver will not look at PR if 2 other members haven't reviewed first
> 
>     * [x]  Did you Have Fun?
> 
> 
> ## Things to Consider
> 
>     * [x]  Even if the above are checked, is this the best way of writing my code?
>       
>       * It's possible to write code that works, but are there ways to make code more efficient?

Can you also put what the pull request does (and why we want to do this) in the description? This makes it easier to review, and if in the future we see this commit and want to know what it is for, it is easier to figure it out.

File: Apps/Src/CAN_Queue.c [hunk 1]
```diff
@@ -20,25 +20,21 @@ static OS_MUTEX canFifo_Mutex;
...
-    CPU_TS ticks;
     RTOS_BPS_MutexCreate(&canFifo_Mutex, "CAN queue mutex");
     OSSemCreate(&canFifo_Sem4,
                 "CAN queue semaphore",
                 0,
                 &err);
     assertOSError(err);
-    OSMutexPend(&canFifo_Mutex, 0, OS_OPT_POST_NONE, &ticks, &err);
-    assertOSError(err);
+    RTOS_BPS_MutexPend(&canFifo_Mutex, 0, OS_OPT_POST_NONE);
```
Comment (ClarkPoon) [inline Apps/Src/CAN_Queue.c:29]: Lmao I just realized we've been calling it like this the whole time and `OS_OPT_POST_NONE` isn't evan a valid option for `OSMutexPend`. See the micrium docs here for the proper options: [https://docs.silabs.com/micrium/latest/micrium-kernel-api/05-kernel-mutex-api#os-mutex-pend](https://docs.silabs.com/micrium/latest/micrium-kernel-api/05-kernel-mutex-api#os-mutex-pend). I think we probably want to use `OS_OPT_PEND_BLOCKING` for everything

File: Tests/Test_WD_Task.c [hunk 2]
```diff
@@ -25,7 +25,7 @@ void Task2(void *p_arg){
         BSP_Light_Toggle(EXTRA);
         OSTimeDly(25, OS_OPT_TIME_DLY, &err);
         // Comment out the following lines to test watchdog timeout
-        OSMutexPend(&WDog_Mutex, 0, OS_OPT_PEND_BLOCKING, NULL, &err);
+        RTOS_BPS_MutexPend(&WDog_Mutex, 0, OS_OPT_PEND_BLOCKING);
```
Comment (ClarkPoon) [inline Tests/Test_WD_Task.c:28]: I think we always pass `0` as the second parameter, so you can remove it and just pass it internally

**Review by ClarkPoon — CHANGES_REQUESTED** [review]

File: Drivers/Src/CANbus.c [hunk 3]
```diff
@@ -119,12 +118,9 @@ static ErrorStatus CANbus_SendMsg(CANId_t id, CANPayload_t payload) {
 	// This is because the software is responsible for
 	// choosing the mailbox to put the message into,
 	// leaving a possible race condition if not protected.
-	OSMutexPend(&CANbus_TxMutex,
-				0,
-				OS_OPT_PEND_BLOCKING,
-				&ts,
-				&err);
-	assertOSError(err);
+	RTOS_BPS_MutexPend(&CANbus_TxMutex,
```
Comment (manthanand) [inline Drivers/Src/CANbus.c:121]: Try to follow coding styles guide by keeping all of them same format.

File: Drivers/Src/CANbus.c [hunk 4]
```diff
@@ -192,16 +188,12 @@ ErrorStatus CANbus_Send(CANId_t id, CANPayload_t payload) {
...
 	OS_ERR err;
 	
 	// The mutex is require to access the CAN receive queue.
-	OSMutexPend(&CANbus_RxMutex,
-				0,
-				OS_OPT_PEND_BLOCKING,
-				&ts,
-				&err);
-	assertOSError(err);
+	RTOS_BPS_MutexPend(&CANbus_RxMutex,
```
Comment (manthanand) [inline Drivers/Src/CANbus.c:194]: Same here.

**Review by manthanand — APPROVED** [review]

**Pull request merged**